File: java/pgwire/src/test/java/com/risingwave/pgwire/duckdb/DuckDbTest.java
package com.risingwave.pgwire.duckdb;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;

import com.risingwave.pgwire.database.Database;
import com.risingwave.pgwire.database.Databases;
import com.risingwave.pgwire.database.PgException;
import com.risingwave.pgwire.database.PgResult;
import com.risingwave.pgwire.types.PgValue;
import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class DuckDbTest {
  @BeforeAll
  static void setup() {
    Databases.setDatabaseManager(new DuckDbManager());
  }

  static List<List<PgValue>> mustGetAllRows(PgResult res) {
    List<List<PgValue>> ret = new ArrayList<>();
    assertDoesNotThrow(
        () -> {
          PgResult.PgIter it = res.createIterator();
          while (it.next()) {
            ret.add(it.getRow());
          }
        });
    return ret;
  }

  @Test
  void runStatementTest() throws PgException {
    Database db = Databases.connect("", "");
    PgResult res = db.runStatement("SELECT 1");
    List<List<PgValue>> rows = mustGetAllRows(res);
    assertEquals(rows.size(), 1);
    assertEquals(rows.get(0).get(0).encodeInText(), "1");
  }
}

File: legacy/pgwire/bin/test/com/risingwave/pgwire/duckdb/DuckDbTest.kt
package com.risingwave.pgwire.duckdb

import com.risingwave.common.exception.PgException
import com.risingwave.pgwire.database.DatabaseManager
import com.risingwave.pgwire.database.PgResult
import com.risingwave.pgwire.types.PgValue
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.ArrayList

internal class DuckDbTest {
  @Test
  @Throws(PgException::class)
  suspend fun runStatementTest() {
    val db = dbManager.connect("", "")
    val res = db.runStatement("SELECT 1")
    val rows = mustGetAllRows(res)
    Assertions.assertEquals(rows.size, 1)
    Assertions.assertEquals(rows[0][0].encodeInText(), "1")
  }

  companion object {
    private var dbManager: DatabaseManager = DuckDbManager()

    fun mustGetAllRows(res: PgResult): List<List<PgValue>> {
      val ret: MutableList<List<PgValue>> = ArrayList()
      Assertions.assertDoesNotThrow {
        val it = res.createIterator()
        while (it.next()) {
          ret.add(it.row)
        }
      }
      return ret
    }
  }
}
